enhancements/network/user-defined-network-segmentation.md

```diff
@@ -357,11 +359,106 @@ block/allow cluster egress or ingress traffic will still be enforced.
+| ExcludeSubnets | List of CIDRs and IP addresses.<br/>IP addresses are removed from the assignable IP address pool and are never passed to the pods.                                                                                                                                                                                                                                                                 | Yes      |
+
+The cluster scoped CRD should have the following additional field:
+
+| Field name | Description                                                                                           | optional |
+|------------|-------------------------------------------------------------------------------------------------------|----------|
+| Selector   | Key-value list of labels used as a selector for which namespace the network should be available for.  | No       |
+
+Since `localnet` topology for primary networks is non-goal, the CRDs controller should reject such requests.
+
+##### status
+The CRD status should reflect the NAD state through conditions.
+For example, when the NAD its created the condition should be placed with status `True`.
+
+For cluster scoped networks, the condition should be true once all desired namespaces are provisioned with the corresponding NAD.
+
+
+
+Granting permissions for users to create OVN networks in namespaces they are allowed should be manged using the RBAC mechanism.
+
+##### Example - Namespace scoped network
+```yaml
+kind: OVNNetwork
+metadata:
+  name: db-network
+  namespace: demo
+spec:
+  topology: layer2
+  role: primary
+  mtu: 9000
+  subnets: ["10.0.0.0/24"]
+  excludeSubnets: ["10.0.0.100"]
+```
+After creation OVN-K finalizer should be added:
+```yaml
+kind: OVNNetwork
+metadata:
+  name: db-network
+  namespace: demo
+  finalizers:
+  - connected-network.k8s.ovn.org
+spec:
+  topology: layer2
+  role: primary
+  mtu: 9000
+  subnets: ["10.0.0.0/24"]
+  excludeSubnets: ["10.0.0.100"]
+status:
+  conditions:
+  - type: "NetworkAttachmentDefinitionReady"
+    status: "True"
+```
+
+##### Example - Cluster scoped network
+```yaml
+kind: ClusterOVNNetwork
+metadata:
+  name: db-network
+  finalizers:
+  - connected-network.k8s.ovn.org
+spec:
+  topology: layer2
+  role: primary
+  selector:
+  - kubernetes.io/metadata.name: "mynamespace"
+  - kubernetes.io/metadata.name: "theirnamespace"
+  mtu: 9000
+  subnets: ["10.0.0.0/24"]
+  excludeSubnets: ["10.0.0.100"]
+status:
+  conditions:
+  - type: "NetworkAttachmentDefinitionReady"
+    status: "False"
+    reason: "NetworkAttachmentDefinitionNotExist"
+    message: "Not all desired namespaces provisioned with NetworkAttachmentDefinition: ['theirnamespace', 'namespace not exist']"
```

Comment:
that sounds wrong to me that a cluster-scoped CR depends on namespaces to be created beforehand?
I think a CR asking for a network between namespaces a,b,c,d should be legit request even if namespaces don't exist.
if they exist then yes we must do the necessary validation
if they don't exist yet, then when they are created we must do the necessary plumbing, that's the way I see it.

Reply:
I think that:
1. if the namespaces do not exist, that should not stop the CR from reporting NetworkReady: true
2. If the namespaces exist, but their NAD has not/cannot be created, NetworkReady: false should be reported

I don't think a reason should be "namespace does not exist".  A selector is just selecting. A namespace that does not exist cannot be selected, which means it does not currently apply to the Network CR and cannot force a failure status.

Reply:
I share @trozet 's opinion.

Reply:
What if we have a request that specify namespaces A, B and C, but non of them exist, should it report NetworkReady=true?

About using selectors,  the more I think about it I realize how it complicate things and see less value in it. see Maybe using plain namespaces is better.

Reply:
Following offline discussion, it was concluded that in case all or at least one namespace is not exist condition should say network is ready.

Thanks for the discussion and explaining :slightly_smiling_face: 
The mentioned example has been fixed.